services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/util/DeviceRegistryUtils.java

```diff
@@ -284,4 +287,36 @@ public static boolean matchesWithClientContext(final JsonObject credential,
                         .orElse(true));
 
     }
+
+    /**
+     * Returns a regex expression based on the given filter value which can be used by the device registry 
+     * implementations to support wildcard matching during search operations.
+     * <p>
+     * The below wildcard characters are supported.
+     * <ol>
+     * <li>`*` will match zero or any number of characters.</li>
+     * <li>`?` will match exactly one character.</li>
+     * </ol>
+     * @param filterValue The value corresponding to the field to use for filtering.
+     * @return The regex expression to use for filtering.
+     * @throws NullPointerException if the filterValue is {@code null}
+     */
+    public static String getRegexExpressionForSearchOperation(final String filterValue) {
+        Objects.requireNonNull(filterValue);
+
+        return Collections.list(new StringTokenizer(filterValue, "*?", true))
+                .stream()
+                .map(token -> (String) token)
+                .map(token -> {
+                    if (token.equals("*")) {
+                        return "(.*)";
```

Comment:
is there a particular reason why you want to use the brackets here?

Reply:
The grouping has no special meaning here. I personally find that it improves the readability in log statements.

Reply:
I would assume that it makes evaluating the regex more expensive because of the grouping, but that might be negligible